### PR TITLE
Add VG for dense map

### PR DESCRIPTION
**Pull request by LBern**
## General
This PR adds a new flag to voxel grid the dense map before we publish it. 

Original dense map
Screenshot from 2020-09-16 01-19-22

Filtered dense map
Screenshot from 2020-09-16 01-23-26

## Changelog
 * added `--vis_pointcloud_filter_dense_map_before_publishing` flag
 * added `--vis_pointcloud_filter_leaf_size_m` flag

### CODE REVIEW

File: visualization/src/resource-visualization.cc [hunk 1]
```diff
@@ -63,6 +64,16 @@ DEFINE_bool(
     vis_pointcloud_color_random, false,
     "If enabled, every point cloud receives a random color.");
 
+DEFINE_bool(
+    vis_pointcloud_filter_dense_map_before_publishing, true,
+    "If enabled, the accumulated dense map will be filtered before "
```
Comment (floriantschopp) [inline visualization/src/resource-visualization.cc:69]: Also works for the non-accumulated one...
```suggestion
    "If enabled, the visualized dense map will be filtered before "
```

File: visualization/src/resource-visualization.cc [hunk 2]
```diff
@@ -304,7 +327,17 @@ void visualizeReprojectedDepthResource(
...
+    pcl::PointCloud<pcl::PointXYZI>::Ptr pcl_cloud(
+        new pcl::PointCloud<pcl::PointXYZI>);
+    CHECK_NOTNULL(pcl_cloud);
+    backend::convertPointCloudType(accumulated_point_cloud_G, &(*pcl_cloud));
+    voxelGridPointCloud(FLAGS_vis_pointcloud_filter_leaf_size_m, pcl_cloud);
+    backend::convertPointCloudType((*pcl_cloud), &ros_point_cloud_G);
+  } else {
+    backend::convertPointCloudType(
+        accumulated_point_cloud_G, &ros_point_cloud_G);
+  }
```
Comment (floriantschopp) [inline visualization/src/resource-visualization.cc:340]: Aren't you doing the same filter twice here (also in line 294-306?), once during the accumulation and once after the accumulation? Does that make sense?

**Review by floriantschopp — APPROVED** [review]
locolgtm

File: visualization/src/resource-visualization.cc (hunk 2 above)
Comment (LBern) [reply visualization/src/resource-visualization.cc:340]: Let me double-check it was late

Comment (LBern) [reply visualization/src/resource-visualization.cc:340]: The line 294 concerns the unaccumulated publisher and from line 329 we would filter the accumulated one. I'll make a tiny fix such that the local tf publisher also benefits from this filtering process, but for the accumulated one we should wait for all the points being added to the global struct.

**Comment by LBern** [comment]
Merging this now for more testing. Need to fix unit tests in `devel/alice`

**Pull request merged**